assets/src/edit-story/elements/video/frame.js

```diff
@@ -14,14 +14,140 @@
  * limitations under the License.
  */
 
+/**
+ * External dependencies
+ */
+import { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { CSSTransition } from 'react-transition-group';
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
 /**
  * Internal dependencies
  */
 import StoryPropTypes from '../../types';
 import MediaFrame from '../media/frame';
+import { elementFillContent } from '../shared';
+import { useStory } from '../../app/story';
+import { ReactComponent as PlayIcon } from '../../icons/play.svg';
+import { ReactComponent as PauseIcon } from '../../icons/pause.svg';
+
+const Play = styled(PlayIcon)`
+  width: 48px;
+  height: 48px;
+`;
+const Pause = styled(PauseIcon)`
+  width: 48px;
+  height: 48px;
+`;
+
+const Wrapper = styled.div`
+  ${elementFillContent}
+`;
+
+const ButtonWrapper = styled.div`
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+  cursor: pointer;
+
+  opacity: 0;
+  &.button-enter {
+    opacity: 0;
+  }
+  &.button-enter-active,
+  &.button-enter-done {
+    opacity: 1;
+    transition: opacity 200ms;
+  }
+  &.button-exit {
+    opacity: 1;
+  }
+  &.button-exit-active,
+  &.button-exit-done {
+    opacity: 0;
+    transition: opacity 200ms;
+  }
+`;
 
 function VideoFrame({ element }) {
-  return <MediaFrame element={element} />;
+  const { id } = element;
+  const [hovering, setHovering] = useState(false);
+  const [isPlaying, setIsPlaying] = useState(false);
+  const {
+    state: { selectedElementIds },
+  } = useStory();
+  const isElementSelected = selectedElementIds.includes(id);
+
+  const onPointerEnter = () => {
+    // Sync UI for auto-play on insert.
+    const videoNode = document.getElementById(`video-${id}`);
+    const currentlyPlaying = videoNode && !videoNode.paused;
+    if (currentlyPlaying && !isPlaying) {
+      setIsPlaying(true);
+    }
+    setHovering(true);
+  };
+
+  useEffect(() => {
+    const videoNode = document.getElementById(`video-${id}`);
+    if (!isElementSelected && videoNode) {
```

Comment:
This effect should also have `if (isElementSelected && video && !video.paused) {setIsPlaying(true);}`. Otherwise, the newly inserted video playback state (`useInsertElement`) will not get picked up here.

Reply:
Right, because that element is only visible on hover, it was not required.
Now added. I also needed to add setTimeout because play (from useInsertElement) happens after this effect. 🙉

Reply:
Ok. Let's proceed. We'll need to come back to it later.